### PR TITLE
Enable NNCF for multihead (hierarchical) classification

### PR DESCRIPTION
**Pull request by JihwanEom**
### Summary:
- Update deep-object-reid commit to support NNCF for multihead classification.
- Add OTE-CLI test for multihead classification, which includes NNCF tests.

### CODE REVIEW

**Comment by JihwanEom** [comment]
This PR will be merged after deep-object-reid PR is merged, and the commit will be updated accordingly. (https://github.com/openvinotoolkit/deep-object-reid/pull/226).

**Comment by harimkang** [comment]
run ote-test

**Comment by JihwanEom** [comment]
run ote-test

**Comment by sovrasov** [comment]
Just a side note: this PR aims to enable NNCF for multilabel classification, but for an unknown reason also contains changes to MPA related to semantic segmentation, which should be reviewed and tested separately.

Does `run ote-test` command work? If it is, where is a link to the corresponding build that this command should trigger?

cc @alexsu52 the tests added are just smoke ones, they don't evaluate the actual result of NNCF quantization.

**Comment by samet-akcay** [comment]
> Just a side note: this PR aims to enable NNCF for multilabel classification, but for an unknown reason also contains changes to MPA related to semantic segmentation, which should be reviewed and tested separately.
> 
> Does `run ote-test` command work? If it is, where is a link to the corresponding build that this command should trigger?
> 
> cc @alexsu52 the tests added are just smoke ones, they don't evaluate the actual result of NNCF quantization.

@sovrasov, `run ote-test` does it is supposed to do, triggering the CI. However, CI doesn't work :/
Here is the [link](https://ci-ote.iotg.sclab.intel.com/job/ote/job/pr-ote-test/) to the CI.

**Review by alexsu52 — COMMENTED** [review]
Hi @JihwanEom, where can I find the results that have been achieved for hierarchical classification models with NNCF?

**Comment by JihwanEom** [comment]
Hi @alexsu52, I'm trying to build a test for MPA hierarchical classification test on CI. However, I found that CI for OTE-CLI test is not working properly. Could you wait a little time?
I'll leave a comment with the link which includes hierarchical NNCF tests when I successfully build.

File: external/deep-object-reid/torchreid_tasks/model_wrappers/classification.py [hunk 1]
```diff
@@ -131,6 +131,9 @@ def softmax_numpy(x: np.ndarray):
 def get_hierarchical_predictions(logits: np.ndarray, multihead_class_info: dict,
                                  pos_thr: float = 0.5, activate: bool = True):
     predicted_labels = []
+    logits_range_dict = multihead_class_info.get('head_idx_to_logits_range', False)
+    if logits_range_dict:  #  json allows only string key, revert to integer.
+        multihead_class_info['head_idx_to_logits_range'] = {int(k):v for k,v in logits_range_dict.items()}
```
Comment (goodsong81) [inline external/deep-object-reid/torchreid_tasks/model_wrappers/classification.py:136]: Is this better than just indexing with str(i)?

Comment (JihwanEom) [reply external/deep-object-reid/torchreid_tasks/model_wrappers/classification.py:136]: Yes, except deploy test cases, the key of 'head_idx_to_logits_range' dictionary incomes as integer. So I designed above code.

**Comment by JihwanEom** [comment]
CI is not working properly, tested CLI tests on local environment with below command:
•	pytest -rs external/model-preparation-algorithm/tests/ote_cli/test_hierarchical_cls_il.py
image
Passed all the test cases.

**Review by goodsong81 — APPROVED** [review]
LGTM.

**Review by supersoob — APPROVED** [review]
LGTM

**Pull request merged**